global:module tests

mb-wali opened the pull request:
# Changes

## Migrated to latest ```pytest-invenio```.
* Removed all the packages that was in the ```tests_require``` section in ```setup.py```, since the ```pytest_invenio``` module has them installed already.
* modified ```pytest.ini``` and ```run-test.sh``` files to the latest ```pytest_invenio```.
* added ```flake8```,```isort``` and ```pycodestyle``` rules to ```setup.cfg```.
This closes #12 .

## Dependencies/packages
* Added some packages to ```setup.py``` that the module needs and uses such as:
  : ```"invenio-rdm-records~=0.20.8",```
  : ```"invenio-search[elasticsearch7]>=1.4.0",```
  : ```"sqlalchemy-continuum>=1.3.11",```

* Removed packages
  : ```"psycopg2>=2.8.4",```
    this package is a driver for postgresql, and in order to use this it is required that postgresql is installed and running.

##  Errors Fixed
### pydocstyle
: D400: First line should end with a period (not 'd')
: D202: No blank lines allowed after function docstring (found 1)
: D210: No whitespaces allowed surrounding docstring text
: D101: Missing docstring in public class
: D107: Missing docstring in __init__
: D102: Missing docstring in public method
: D209: Multi-line docstring closing quotes should be on a separate line
: D205: 1 blank line required between summary line and description (found 0)
: D100: Missing docstring in public module
: D103: Missing docstring in public function
### Flake
: W291 trailing whitespace
: W503 line break before binary operator
: E266 too many leading '#' for block comment
: E713 test for membership should be 'not in'
: E712 comparison to False should be 'if cond is False:' or 'if not cond:'


### Cannot parse file. [ ISSUE_a ]
**1.**
      ```
      WARNING: Error in file invenio_rdm_pure/source/pure/import_records.py: Cannot parse file. 
      ```
      This error was in function ```_add_attribute``` starting lines [283]. For now commented this function.

**2.**
      ```
      WARNING: Error in file invenio_rdm_pure/source/rdm/add_record.py: Cannot parse file. 
      ```
      This error was in function ```_add_field_sub``` starting lines [377]. For now commented this function.

**3.**
      ```
      WARNING: Error in file invenio_rdm_pure/source/rdm/run/groups.py: Cannot parse file.  
      ```
      This error was in somewhere from line [241] to [562]. For now commented these lines.

### Imports and mismatch file names.
The module had few a lot of imports, and naming mismatch through out the module.
* changes in imports :
   FROM -> ```from source.reports import Reports``` TO -> ```from invenio_rdm_pure.source.reports import Reports```.
**This change was necessary for module in order to find the files inside the virtual environments.** [ ISSUE_b ]
 
* File name mismatches.
These following files have been renamed:
**FROM** ```invenio_rdm_pure/source/pure/requests.py``` **TO** ```invenio_rdm_pure/source/pure/requests_pure.py ```.
**FROM** ```invenio_rdm_pure/source/rdm/requests.py``` **TO** ``` invenio_rdm_pure/source/rdm/requests_rdm.py  ```.
**FROM** ```invenio_rdm_pure/source/general_functions.py``` **TO** ```invenio_rdm_pure/source/general_functions_source.py ```.
**FROM** ```invenio_rdm_pure/source/pure/general_functions.py``` **TO** ```invenio_rdm_pure/source/pure/general_functions_pure.py```.
**FROM** ```invenio-rdm-pure/invenio_rdm_pure/source/rdm/run/uuid.py``` **TO** ```invenio-rdm-pure/invenio_rdm_pure/source/rdm/run/uuid_run.py```.
**These renaming was done to fix the ```E   _pytest.pathlib.ImportPathMismatchError: ``` error.**

### indexError: list index out of range
There was an index out of range error in ```ext.py``` [HERE](https://github.com/tu-graz-library/invenio-rdm-pure/blob/9278fce27cc72b4af4d9353b3b6ef80ad7feb71a/invenio_rdm_pure/ext.py#L46) - **temp-fix:** changed all indexes to use [0].

#### TODO:
**Create issues.**
**[ ISSUE_a ]** Cannot parse file. "un-comment and debug to find the errors."
**[ ISSUE_b ]** Fix module imports. "fix the way its importing files in this module."
**[ ISSUE_c ]** Replace "Descriptions & File Descriptions with actual comments."

Commits from @bengaal which closes #18 and #19.

mb-wali committed:
global:tests

* migrated to pytest-invenio
* global:docstring readthedocs fixed
* Cannot parse file - some function in several files has a error. commented for now.
* install_requires & test_requires added packages that this module needs.

bengaal committed:
Corrected import statements to use local paths

mb-wali committed:
global:imports & removed from compile

* commented some functions and files that are not necessary to be compile.
* fixed:imports from folder to module name import
* fixed mismatch file nameing, with renaming the files and sorting the imports.

mb-wali committed:
global:mismatch naming & imports

*fixed:mismatch naming for python files and thier imports.
* E266 docsttings
* W503 line break before binary operator
* E713 test for membership should be 'not in'
E712 comparison to False should be 'if cond is False:' or 'if not cond:'
* IndexError: list index out of range in ext.py

utnapischtim commented on invenio_rdm_pure/source/reports.py:
```diff
@@ -1,63 +1,39 @@
 
 from setup import dirpath, log_files_name
 
-report_templates = {
```
why removing report_templates. there is a reference to it in line 30

mb-wali replied:
I will check this.

utnapischtim commented on pytest.ini:
```diff
@@ -6,6 +6,6 @@
-testpaths = docs tests invenio_rdm_pure
+addopts = --isort --pydocstyle --pycodestyle --doctest-glob="*.rst" --doctest-modules --cov=invenio_rdm_pure --cov-report=term-missing
+testpaths = tests invenio_rdm_pure
+live_server_scope = module
```
please add a newline add the end of the file

mb-wali replied:
Why a new line is necessary?

utnapischtim replied:
it is best described in this link: https://stackoverflow.com/a/729795

utnapischtim commented on run-tests.sh:
```diff
@@ -6,8 +6,13 @@
+
+tests_exit_code=$?
+docker-services-cli down
+exit "$tests_exit_code"
```
please add a newline add the end of the file

utnapischtim commented on setup.py:
```diff
@@ -15,13 +15,8 @@
-    "pytest-pep8>=1.0.6",
-    "pytest-invenio>=1.2.1",
+    "pytest-invenio>=1.4.0",
+     "invenio-search[elasticsearch7]>=1.4.0",
```
there is a white space to many

mb-wali committed:
review:commits

* new line for txt files
* report_templates was removed, in testing added again.
* extra space removed in setup.py.

bengaal commented on invenio_rdm_pure/ext.py:
```diff
@@ -35,19 +35,17 @@ def init_config(self, app):
             ):
                 app.config.setdefault(k, getattr(config, k))
         if "SQLALCHEMY_DATABASE_URI" in app.config:
             database_uri = app.config.get("SQLALCHEMY_DATABASE_URI").split("//")[1]
```
Why is it okay to access the second element here, but not in the rest of this part?

mb-wali replied:
@bengaal -  I think there is no expected value as ```SQLALCHEMY_DATABASE_URI=postgresql+psycopg2://username:inveniordm@127.0.0.1:5432/invenio-app-rdm```. And the fact that it does not return error on the first or second element - only logical explanation would be that there is a default value assigned to the variable and that value does not have other indexes as we expect. And as far as I remember it should use a default value as such ```SQLALCHEMY_DATABASE_URI=sqlite://```.

bengaal replied:
The problem isn't present anymore with the original version. So I will reset these array accesses.

bengaal committed:
uuid_run.py Added missing docstring and header to module

bengaal committed:
Sorted imports

bengaal commented on setup.py:
```diff
@@ -43,8 +38,9 @@
     "invenio_oauthclient>=1.2.1",
+    "invenio-rdm-records~=0.20.8",
+    "sqlalchemy-continuum>=1.3.11",
 ]
```
please also add `"invenio-records-rest~=1.7.2"`, otherwise we get the following error: 
```Warning, treated as error:
autodoc: failed to import module 'ext' from module 'invenio_rdm_pure'; the following exception was raised:
No module named 'invenio_records_rest'

mb-wali replied:
No need - it's already installed.
```invenio-rdm-records```installs  ```invenio-records-files``` see [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/05dfbfdcfc94fbbbf5fadddb01c8b5f9659ba58c/setup.py#L75), then this module installs ```invenio-records-rest``` see [Here](https://github.com/inveniosoftware/invenio-records-files/blob/1546ef4517816f12d3000b7d36c2cf9a422f4541/setup.py#L61).
all connected : )
If you do a pip list in your virtual env you can see that this module is installed.

I think the error is from [sphinx](https://github.com/sphinx-doc/sphinx).

bengaal committed:
global Added __init__.py files

bengaal committed:
Added Black formatting

bengaal committed:
Corrected import statements

bengaal committed:
setup.cfg Added ignore flags

Added ignore flags in order to support Black formatting.

bengaal committed:
ext.py Reset array accesses

The array accessses were reported as erroneous before, however
this is not present anymore. Logically and for the sake of automated
setup/configuration these are correct.

bengaal committed:
Resolved cannot parse file error

Uncommented the parts causing parsing errors.
The problem was the indentation of the method docstrings.
Indenting them one level deeper resolved the errors.
